pfs_middleware/pfs_middleware/middleware.py

```diff
@@ -402,7 +417,13 @@ def extract_object_metadata_from_headers(headers):
         if (header.startswith("X-Object-Meta-") or
                 header.startswith("X-Object-Sysmeta-") or
                 header in SPECIAL_OBJECT_METADATA_HEADERS):
-            meta_headers[header] = value
+
+            # do not let a client pass in ORIGINAL_MD5_HEADER
+            if (header != ORIGINAL_MD5_HEADER and
+                header != ORIGINAL_MD5_HEADER.title()):
```

Comment:
I think `tox -e lint` will be happier with something like
```
if header != ORIGINAL_MD5_HEADER and \
        header != ORIGINAL_MD5_HEADER.title():
    meta_headers[header] = value
```
though maybe it'd be more pythonic as
```
if header not in (ORIGINAL_MD5_HEADER,
                  ORIGINAL_MD5_HEADER.title()):
    meta_headers[header] = value
```
? The reduced repetition might even let it all fit on one line.